Doc/library/pathlib.rst

```diff
@@ -1633,6 +1633,34 @@ Copying, renaming and deleting
       Added return value, return the new :class:`!Path` instance.
 
 
+.. method:: Path.delete(ignore_errors=False, on_error=None)
+
+   Recursively remove this file or directory tree.
+
+   If *ignore_errors* is true, errors resulting from failed removals will be
+   ignored. If *ignore_errors* is false or omitted, and a function is given to
+   *on_error*, it will be called each time an exception is raised. If neither
+   *ignore_errors* nor *on_error* are supplied, exceptions are propagated to
+   the caller.
+
+   If the optional argument *on_error* is specified, it should be a callable;
+   it will be called with one argument of type :exc:`OSError`. The
+   callable can handle the error to continue the deletion process or re-raise
+   it to stop. Note that the filename is available as the
+   :attr:`~OSError.filename` attribute of the exception object.
```

Comment:
This feels duplicative as we almost entirely outline *on_error* (a function taking an exception) in the first paragraph, and go on to rehash much of it in the second. However, I tried a sketch rewrite on GH's web interface and didn't get far. 

The best I had was changing the first para to say "If *ignore_errors* is false or omitted and *on_error* is given, that function will determine the error behaviour, as described below", but it's not great.

```suggestion
   If *ignore_errors* is true, errors resulting from failed deletions will be
   ignored. If *ignore_errors* is false or omitted, and a function is given to
   *on_error*, it will be called each time an exception is raised. If neither
   *ignore_errors* nor *on_error* are supplied, exceptions are propagated to
   the caller.

   If the optional argument *on_error* is specified, it should be a callable;
   it will be called with one argument of type :exc:`OSError`. The
   callable can handle the error to continue the deletion process or re-raise
   it to stop. Note that the filename is available as the
   :attr:`~OSError.filename` attribute of the exception object.
```

Reply:
Agreed. Better now?

FWIW I'm considering removing the *ignore_errors* and *on_error* arguments in a subsequent PR, at least for a while, because they're underdesigned and unnecessary for most use cases. So these paragraphs might not be around for long...

Reply:
I think ``ignore_errors`` has merit in the case of "I want to try and clean up this folder, but it isn't critical", but that could be handled explicitly by the user with a try-except. The real use is for the per-error callback, but that could perhaps be handled with an `ExceptionGroup` nowadays?

Reply:
I agree we'll probably need to support _some_ kind of error handling customisation, but I'm not sure exactly what it looks like. I'll see if `ExceptionGroup` fits!